packages/docs/src/helpers/urlUtils.ts

```diff
@@ -26,7 +26,12 @@ export function makeStorybookUrl(storyId, theme) {
 }
 
 export function makePageUrl(fileRelativePath, location: LocationInterface) {
-  return join('/', fileRelativePath.replace('.mdx', ''), '/', location.search);
+  return join(
+    '/',
+    fileRelativePath.replace('index.mdx', '').replace('.mdx', ''),
```

Comment:
Make sure the URL for `/components/text-field/` is not `/components/text-field/index/`. Even though I changed how the links were organized in the nav, I didn't want to actually change the URL for our text field component page. This is also more succinct than either `components/text-field/text-field/` or `/components/text-field/index/`.